File: user-service/src/main/java/socialapp/userservice/model/entity/Friendship.java
package socialapp.userservice.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import socialapp.userservice.model.enums.RelationStatus;

@Entity
@Getter
@Setter
public class Friendship extends Relation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Override
    public RelationStatus getRelationStatus() {
        return RelationStatus.FRIENDSHIP;
    }
}

File: user-service/src/main/kotlin/socialapp/ktuserservice/model/entity/Friendship.kt
package socialapp.ktuserservice.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import socialapp.ktuserservice.model.entity.BaseEntity
import socialapp.ktuserservice.model.enums.RelationStatus

@Entity
class Friendship: Relation() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
}